src/utilities/effects.ts

```diff
@@ -285,19 +285,39 @@ const effects = {
       checkConstraintsStatus.set(Status.Incomplete);
       if (plan !== null) {
         const { id: planId } = plan;
-        const data = await reqHasura<ConstraintResult[]>(
+        const data = await reqHasura<ConstraintResponse[]>(
           gql.CHECK_CONSTRAINTS,
           {
             planId,
           },
           user,
         );
+        const { ConstraintResponses } = data;
+        if (ConstraintResponses) {
+          constraintResponse.set(ConstraintResponses);
 
-        const { constraintResults } = data;
-        if (constraintResults != null) {
-          constraintResultsResponse.set(constraintResults);
-          checkConstraintsStatus.set(Status.Complete);
-          showSuccessToast('Check Constraints Complete');
+          // find only the constraints compiled.
+          const successfulConstraintResults: ConstraintResult[] = ConstraintResponses.filter(
+            constraintResponse => constraintResponse.success,
+          ).map(constraintResponse => constraintResponse.results);
+
+          const failedConstraintResponses = ConstraintResponses.filter(
+            constraintResponse => !constraintResponse.success,
+          );
+
+          if (successfulConstraintResults.length != ConstraintResponses.length) {
+            showFailureToast('Partial Constraints Completed');
+            failedConstraintResponses.forEach(failedConstraint => {
+              failedConstraint.errors.forEach(error => {
+                catchError(`${error.message}`, error.stack);
+              });
+            });
+
+            checkConstraintsStatus.set(successfulConstraintResults.length !== 0 ? Status.Incomplete : Status.Failed);
+          } else {
+            showSuccessToast('All Constraints Completed');
```

Comment:
should these toasts mention something about "checking" since the button to trigger this is "Check Constraints"?